pallets/mt/src/lib.rs

```diff
@@ -277,24 +293,29 @@ pub mod pallet {
 			})
 		}
 
-		#[pallet::weight(0)]
+		#[pallet::weight(T::WeightInfo::force_set_maintainer())]
 		pub fn force_set_maintainer(origin: OriginFor<T>, new_maintainer: T::AccountId) -> DispatchResultWithPostInfo {
 			T::ForceOrigin::ensure_origin(origin)?;
 			// set the new maintainer
 			Maintainer::<T, I>::try_mutate(|maintainer| {
 				*maintainer = new_maintainer.clone();
-				Self::deposit_event(Event::MaintainerSet(Default::default(), T::AccountId::default()));
+				Self::deposit_event(Event::MaintainerSet(Default::default(), new_maintainer));
 				Ok(().into())
 			})
 		}
 
-		#[pallet::weight(0)]
+		#[pallet::weight(T::WeightInfo::force_set_default_hashes(default_hashes.len() as u32))]
 		pub fn force_set_default_hashes(
 			origin: OriginFor<T>,
 			default_hashes: Vec<T::Element>,
 		) -> DispatchResultWithPostInfo {
 			T::ForceOrigin::ensure_origin(origin)?;
-			// set the new maintainer
+			let len_of_hashes = default_hashes.len();
```

Comment:
You should ensure here, also, that the `len_of_hashes == T::MaxTreeDepth::get()`.

Reply:
Well, the default hashes len is equal to the MaxTreeDepth.
So, I don't see any need for `MaxDefaultHashes` at all.

Reply:
Okay, I'll make that fix